src/transactions/ChangeTrustOpFrame.cpp

```diff
@@ -218,21 +231,24 @@ ChangeTrustOpFrame::doApply(AbstractLedgerTxn& ltx)
                 auto const& cp =
                     mChangeTrust.line.liquidityPool().constantProduct();
 
-                managePoolOnDeletedTrustLine(
-                    ltx, cp, tlEntry().asset.liquidityPoolID(), getSourceID());
+                // managePoolOnDeletedTrustLine will create a child LedgerTxn
+                // and the trustLine will be deactivated, so do what we need
+                // before.
+                auto poolID = tlEntry().asset.liquidityPoolID();
+                trustLine.erase();
+
+                managePoolOnDeletedTrustLine(ltx, cp, poolID, getSourceID());
```

Comment:
So I reworked this a bit. It's cleaner now and the nested `LedgerTxn` is always created. This isn't the most optimal approach since the nested `LedgerTxn` is created unnecessarily sometimes, but it's less error prone.

Reply:
This approach is great, much less error prone. The same problem exists at the call-site of `tryManagePoolOnNewTrustLine` (line 296), would it be possible to do something similar there?

Reply:
Done. Squashed all existing existing fixup commits and added one more for this change.